validator/src/services/mixmining/mod.rs

```diff
@@ -1 +1,163 @@
+use db::MixminingDb;
+use models::*;
+
+pub mod db;
 pub mod health_check_runner;
+pub mod models;
+mod tests;
+
+pub struct Service {
+    db: MixminingDb,
```

Comment:
From what I see, everything using `Service`, has it behind `Arc<Mutex<Service>>` and have to deal with locks, etc. Now, note that the next suggestion has a lot of personal bias, because I've been using the following pattern a lot and I'm still not entirely sure whether it's a correct approach, but I've been seeing it in `mio`, `tokio` and `futures`, so it might not be as bad.

What if `Service` was responsible for dealing with the locks and synchronization itself, making all callers oblivious to them, similarly to what I've done with the client ledger?
https://github.com/nymtech/nym/blob/develop/sfw-provider/src/provider/client_handling/ledger.rs#L35

Reply:
So say `Service::add(...)` would deal with the inner lock rather than whoever is calling it